compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -387,3 +388,37 @@ ireeGPUGetSingleSubgroupLayout(MlirAttribute attr, uint32_t fragment) {
   result.element = wrap(builder.getI64ArrayAttr(layout.element));
   return result;
 }
+
+ireeGPUTargetInfo
+ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr) {
+  assert(!mlirAttributeIsNull(attr) && "attr cannot be null");
+  auto executableTargetAttr =
+      llvm::cast<mlir::iree_compiler::IREE::HAL::ExecutableTargetAttr>(
+          unwrap(attr));
+
+  assert(executableTargetAttr && "attr is not a HAL::ExecutableTargetAttr");
+
+  ireeGPUTargetInfo targetInfo = {};
+  mlir::MLIRContext *context = executableTargetAttr.getContext();
+  mlir::iree_compiler::IREE::GPU::TargetAttr gpuTargetAttr =
+      mlir::iree_compiler::getGPUTargetAttr(context, executableTargetAttr);
+
+  if (gpuTargetAttr) {
```

Comment:
Use an early return instead: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code